Examples/ResourceLeak/README.md

```diff
@@ -0,0 +1,46 @@
+## Procedure-local Bugs of Resource Leaks
+
+**CSharpCodeAnalyzer** is able to detect resource leaks which include the C# scenarios outlined below. For further detail on its capabilities, see [here](http://www.eecs.qmul.ac.uk/~ddino/papers/nasa-infer.pdf). 
+
+## Supported scenarios
+
+### 1. Deallocation of IDisposable Local Variables: 
+
+Instances of classes deriving from IDisposable (such as input streams, output streams, http connections, and cursors) create resource leaks if they are not closed or disposed of.
+
+```c#
+public void ResourceLeakBad(){
+    StreamReader sr = new StreamReader("whatever.txt");  
+    string data = sr.ReadToEnd();
+    Console.WriteLine(data);
+    // FIXME: should close the stream by calling sr.Close().
+}
+```
+							
+### 2. Resource Allocation in Method Invocation: 
+When a resource allocation is included as an argument to a constructor/method invocation, if this constructor/method invocation fails it can leave an an unreachable resource that no one can dispose.
+	
+The following example 
+```c#
+var gzipStream = new GZipStream(new FileStream(out, FileMode.Create), CompressionMode.Compress);
+```
+is bad in case the outer constructor, `GZipStream`, throws an exception. In that case, no one will have a hold of the `FileStream` and so no one will be able to dispose it.
+	
+### 3. Resource Allocation inside libraries:
+Some resources are created inside libraries instead of by "new". For instance,
```

Comment:
As I mentioned in a previous reply, I used the official phrase from Infer team. So for this paragraph I rephrased as:

"
Some resources are created inside libraries instead of by "new". For instance,
```c#
ICursor cursor = SQLiteDatabase.Query(…)
```
allocated `ICursor` resource will leak if it is not disposed.
"